PURL-TYPES.rst

```diff
@@ -521,6 +521,44 @@ swift
       pkg:swift/github.com/Alamofire/Alamofire@5.4.3
       pkg:swift/github.com/RxSwiftCommunity/RxFlow@2.12.4
 
+vcpkg
+-----
+``vcpkg`` for `vcpkg <https://vcpkg.io>`_ C and C++ packages:
+
+- By default, vcpkg will build C and C++ dependencies from source, using configurations compatible with the user's project. For this and other reasons, it uses different terminology, documented here.
+
+  - **port** - A package, along with it's build scripts and possibly minor modifications.
+  - **registry** - A collection of ports, possibly private to the user. Analogous to **repository**.
+
+- ``namespace``: Currently reserved for future use and should be empty
+
+- ``name``: The case-sensitive name of the port. **Required**
+- ``version``: The port version. **Required**
+- ``qualifiers``: The qualifiers below are used to provide more specific information on a port's origin registry and to distinguish between multiple instances of the same port dependency within the same project (for example, when the project targets multiple platforms).
+
+  - ``repository_url``: A `percent-encoded <https://www.rfc-editor.org/rfc/rfc3986#section-2.1>`_ absolute URI for the registry the port came from. For filesystem registries, the URI will have a `file` URI scheme. **Optional**
```

Comment:
You may want to leave out encoding issues from here. This is handled in the main spec also `repository_url` is already specified there.

You could start in the same way as other types with something along these lines?

```
The default repository is ``https://github.com/microsoft/vcpkg``.Use ``repository_url`` for an alternative port registry; for filesystem registries, use a `file:///` URI.
```

Reply:
Removed the part about the encoding, as it is not specific to vcpkg